### PR TITLE
NOISSUE: Add Domains to the SDK

### PR DESCRIPTION
**Pull request by arvindh123**
Pull request title should be `MG-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](https://github.com/absmach/magistrala/blob/master/CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
- Provides SDK for Domains

### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### CODE REVIEW

File: cmd/test/main.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package main
```
Comment (ianmuchyri) [inline cmd/test/main.go:1]: remove test file from cmd

File: cli/domains.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+			l, err := sdk.ListDomainUsers(args[1], pageMetadata, args[2])
+			if err != nil {
+				logError(err)
+				return
+			}
+			logJSON(l)
+
+		},
+	},
```
Comment (ianmuchyri) [inline cli/domains.go:104]: we need a cli function for list user's domains

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

File: cli/domains.go (hunk 2 above)
Comment (arvindh123) [reply cli/domains.go:103]: Added in users

File: cmd/test/main.go (hunk 1 above)
Comment (arvindh123) [reply cmd/test/main.go]: removed

File: cli/domains.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+// NewDomainsCmd returns domains command.
+func NewDomainsCmd() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "domains [create | get | update | enable | disable | enable | users | assign | unassign]",
+		Short: "Domains management",
+		Long:  `Domains management: create, update, retrieve domains , assign / unassign users to domains and list users of domain"`,
```
Comment (dborovcanin) [inline cli/domains.go:225]: `assign/unassign`

File: pkg/sdk/go/sdk.go [hunk 4]
```diff
@@ -958,6 +960,113 @@ type SDK interface {
...
+	EnableDomain(domainID, token string) errors.SDKError
+
+	// DisableDomain changes the status of the domain to disable.
+	//
+	// example:
+	//  err := sdk.DisableDomain("domainID", "token")
+	//  fmt.Println(err)
+	DisableDomain(domainID, token string) errors.SDKError
+
+	// AddUserToDomain add user to a domain.
```
Comment (ianmuchyri) [inline pkg/sdk/go/sdk.go:1049]:
```suggestion
	// AddUserToDomain adds a user to a domain.
```

File: pkg/sdk/go/sdk.go [hunk 5]
```diff
@@ -958,6 +960,113 @@ type SDK interface {
...
+	// example:
+	// req := sdk.UsersRelationRequest{
+	//		Relation: "viewer", // available options: "owner", "admin", "editor", "viewer",  "member"
+	//  	UserIDs: ["user_id_1", "user_id_2", "user_id_3"]
+	// }
+	// err := sdk.AddUserToDomain("domainID", req, "token")
+	// fmt.Println(err)
+	AddUserToDomain(domainID string, req UsersRelationRequest, token string) errors.SDKError
+
+	// RemoveUserFromDomain remove user from a domain.
```
Comment (ianmuchyri) [inline pkg/sdk/go/sdk.go:1060]:
```suggestion
	// RemoveUserFromDomain removes a user from a domain.
```

File: pkg/sdk/go/sdk.go [hunk 6]
```diff
@@ -958,6 +960,113 @@ type SDK interface {
...
+	ListUserDomains(userID string, pm PageMetadata, token string) (DomainsPage, errors.SDKError)
+
+	// EnableDomain changes the status of the domain to enabled.
+	//
+	// example:
+	//  err := sdk.EnableDomain("domainID", "token")
+	//  fmt.Println(err)
+	EnableDomain(domainID, token string) errors.SDKError
+
+	// DisableDomain changes the status of the domain to disable.
```
Comment (ianmuchyri) [inline pkg/sdk/go/sdk.go:1042]:
```suggestion
	// DisableDomain changes the status of the domain to disabled.
```

File: pkg/sdk/go/domains.go [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	url := fmt.Sprintf("%s/%s", sdk.domainsURL, domainsEndpoint)
+
+	_, body, sdkerr := sdk.processRequest(http.MethodPost, url, token, data, nil, http.StatusOK)
+	if sdkerr != nil {
+		return Domain{}, sdkerr
+	}
+
+	do := Domain{}
```
Comment (dborovcanin) [inline pkg/sdk/go/domains.go:42]:
```GO
var d Domain
```

File: pkg/sdk/go/domains.go [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+func (sdk mgSDK) ListDomainUsers(domainID string, pm PageMetadata, token string) (UsersPage, errors.SDKError) {
+	url, err := sdk.withQueryParams(sdk.usersURL, fmt.Sprintf("%s/%s/%s", domainsEndpoint, domainID, usersEndpoint), pm)
+	if err != nil {
+		return UsersPage{}, errors.NewSDKError(err)
+	}
+	_, body, sdkerr := sdk.processRequest(http.MethodGet, url, token, nil, nil, http.StatusOK)
+	if sdkerr != nil {
+		return UsersPage{}, sdkerr
+	}
+	up := UsersPage{}
```
Comment (dborovcanin) [inline pkg/sdk/go/domains.go:113]: The same, `var up UserPage`, same for other variable declarations.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by ianmuchyri — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**